File: Practica01/Cronometro.java
public class Cronometro{
  private long inicia,finaliza;
  public Cronometro(){
    inicia=System.currentTimeMillis();
  }
  public long getInicia() {
        return inicia;
  }
  public long getFinaliza() {
        return finaliza;
  }
  public void inicia() {
        inicia=System.currentTimeMillis();
  }
  public void detener() {
        finaliza=System.currentTimeMillis();
  }
  public long lapsoDeTiempo() {
        return finaliza-inicia;
  }
    public static void main(String[] args) {
      int[] numeros=new int[100000];
      for (int i=0; i<numeros.length; i++) {
            numeros[i]=(int)(Math.random()*100000);
      }
      Cronometro c=new Cronometro();
      c.inicia();
      for (int i=0; i<numeros.length-1;i++) {
            int menor=i;
            for (int j=i+1; j<numeros.length; j++) {
                if (numeros[j]<numeros[menor]) {
                    menor=j;
                }
            }
            int aux=numeros[i];
            numeros[i]=numeros[menor];
            numeros[menor]=aux;
      }
      c.detener();
      System.out.println("Tiempo: "+c.lapsoDeTiempo()+" milisegundos");
    }
    }

File: Practica01/Cronometro.py
import time, random;
class Cronometro:
    def __init__(self):
        self.inicia=time.time()*1000
        self.finaliza=0
    def getInicia(self):
        return self.inicia
    def getFinaliza(self):
        return self.finaliza
    def iniciaCronometro(self):
        self.inicia=time.time()*1000
    def detener(self):
        self.finaliza=time.time()*1000
    def lapsoDeTiempo(self):
        return self.finaliza-self.inicia
numeros=[0]*1000
for i in range(len(numeros)):
    numeros[i]=random.randint(0,99999)
c=Cronometro()
c.iniciaCronometro()
for i in range(len(numeros)- 1):
    menor=i 
    for j in range(i+1, len(numeros)):
        if numeros[j]<numeros[menor]:menor=j
    aux=numeros[i] 
    numeros[i]=numeros[menor] 
    numeros[menor]=aux
c.detener()
print("Tiempo: ",c.lapsoDeTiempo(), " milisegundos")